Sources/Core/AWSClientRuntime/Tests/AWSClientRuntimeTests/Protocols/AWSJSON/AWSJSONErrorTests.swift

```diff
@@ -0,0 +1,119 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import SmithyTestUtil
+import SmithyHTTPAPI
+import Smithy
+@_spi(SmithyReadWrite) import class SmithyJSON.Reader
+@_spi(SmithyReadWrite) import struct AWSClientRuntime.AWSJSONError
+@_spi(SmithyReadWrite) import enum ClientRuntime.BaseErrorDecodeError
+import XCTest
+
+class AWSJSONErrorTests: HttpResponseTestBase {
+    // These error codes are taken from the examples in
+    // https://smithy.io/2.0/aws/protocols/aws-json-1_0-protocol.html#operation-error-serialization
+    // (one extra case with leading & trailing whitespace is added)
+    let errorCodes = [
+        "FooError",
+        "   FooError  ",
+        "FooError:http://internal.amazon.com/coral/com.amazon.coral.validate/",
+        "aws.protocoltests.restjson#FooError",
+        "aws.protocoltests.restjson#FooError:http://internal.amazon.com/coral/com.amazon.coral.validate/"
+    ]
+
+    // MARK: - error code decoding & sanitization
```

Comment:
Tests are added below to ensure that error code can be decoded no matter which of the allowed places it is located.